From f8ac94525280f933cb600fa8f26cef6091389bcd Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Fri, 19 Jan 2024 09:14:09 +0530
Subject: [PATCH] arm64: dts: st: Add wakeup source for usbh using exti on
 stm32mp25

Remote-wakeup capable device connected to USBH can act as a
source to wakeup system from low-power state on mp25 platform
Enable exti-intr connected to utmi_suspend_o_n output of USB2H
as wakeup-source

Change-Id: Id85ab9b08149327fa9a1409d1d081465628282da
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/370803
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index f8303a3e1438..fc6f5447395f 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2264,6 +2264,8 @@ usbh: usb@482e0000 {
 				ranges = <0x482e0000 0x482e0000 0x20000>;
 				access-controllers = <&rifsc 63>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
+				interrupts-extended = <&exti1 43 IRQ_TYPE_EDGE_RISING>;
 				status = "disabled";
 
 				usbh_ohci: usb@482e0000 {
@@ -2274,6 +2276,7 @@ usbh_ohci: usb@482e0000 {
 					interrupts = <GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>;
 					phys = <&usb2_phy1>;
 					phy-names = "usb";
+					wakeup-source;
 				};
 
 				usbh_ehci: usb@482f0000 {
@@ -2285,6 +2288,7 @@ usbh_ehci: usb@482f0000 {
 					companion = <&usbh_ohci>;
 					phys = <&usb2_phy1>;
 					phy-names = "usb";
+					wakeup-source;
 				};
 			};
 
-- 
2.39.5

